src/Ormolu/Imports.hs

```diff
@@ -44,9 +44,11 @@ compareIdecl (L _ m0) (L _ m1) =
     n1 = unLoc (ideclName m1)
     isPrelude = (== "Prelude") . moduleNameString
 
--- | Sort located import or export.
+-- | Sort located import or export, dropping duplicates.
 sortLies :: [LIE GhcPs] -> [LIE GhcPs]
-sortLies = sortBy (compareIE `on` unLoc) . fmap (fmap sortThings)
+sortLies = nubSortBy (compareIE `on` unLoc) . fmap (fmap sortThings)
+  where
+    nubSortBy f = (>>= take 1) . groupBy (\a b -> f a b == EQ) . sortBy f
```

Comment:
This doesn't read very nice to me. Why does `nubSortBy` take an argument when it's only used once? Furthermore, what's wrong with doing `sortBy` and then `nubBy` (available in `base`)?

Reply:
The argument is used twice, or do I misunderstand?

`nubBy` is O(n^2), whereas this should fuse away. I suppose it shouldn't matter for the sizes we're dealing with though, would you rather use `nubBy`?

Reply:
Performance hardly matters here at all, while readability does. The bottleneck of Ormolu is parsing, it is incomparably slower.

Reply:
Makes sense. How's this?